config.go

```diff
@@ -0,0 +1,158 @@
+package kqueuey
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+var (
+	TestFlag    uint
+	SearchPaths = []string{SearchPath0, SearchPath1, SearchPath2, SearchPath3}
+
+	errConfigFileNotLocated   = fmt.Errorf("config file not found, valid paths: %v", SearchPaths)
+	errDuplicateStoragePath   = errors.New("storage path must be unique per node to avoid file locking conflicts")
+	errDuplicateNodeID        = errors.New("node id must be unique per node to conform to rafts consensus protocol")
+	errDuplicateNodePort      = errors.New("node port most be unique to allow proper communication between nodes")
+	errNodeIdNotFound         = errors.New("node id not found")
+	errNodeStorageDirNotFound = errors.New("node storage directory not found")
+	errUnknownNodeAddress     = errors.New("unknown node address")
+	errRaftClusterIdNotFound  = errors.New("cluster id was not found")
+)
+
+const (
+	SearchPath0 = "/config/"
+	SearchPath1 = "/etc/config/"
+	SearchPath2 = "/var/lib/config/"
+	SearchPath3 = "/data/"
+
+	ConfigType     = "yaml"
+	ConfigFileName = "kqueuey-config"
+)
+
+type Configuration struct {
+	BadgerOpts BadgerDBConfig `mapstructure:"storage"`
+	RaftOpts   RaftConfig     `mapstructure:"raft"`
+}
+
+type BadgerDBConfig struct {
+	NumCompactors   int    `mapstructure:"num_compactors"`
+	CompressionType string `mapstructure:"compression_type"`
+	SyncWrites      bool   `mapstructure:"sync_writes"`
+}
+
+type RaftConfig struct {
+	ClusterID string       `mapstructure:"cluster_id"`
+	Nodes     []NodeConfig `mapstructure:"nodes"`
+}
+
+type NodeConfig struct {
+	Id         string `mapstructure:"id"`
+	BindAddr   string `mapstructure:"bind_addr"`
+	StorageDir string `mapstructure:"storage_dir"`
+}
+
+func LoadConfiguration() (*Configuration, error) {
+	v := initializeViper()
+
+	if err := v.ReadInConfig(); err != nil {
+		return nil, err
+	}
+
+	if v.ConfigFileUsed() == "" {
+		return nil, errConfigFileNotLocated
+	}
```

Comment:
Prolly will remove this, under the hood `ReadInConfig` already check what full config path should be used, and throws an error if no config path can be used